add configurable EEPROM length

oclyke opened the pull request:
since the flashContent buffer is declared on the stack it may be dynamically sized the user can take advantage of this to allow for any size EEPROM up to the allocated 1 page of flash

- added 'allowedSize' member to EEPROMClass
- added setter for allowedSize 'setLength'
- adjusted 'length' to return allowedSize

- adjusted 'writeBlockToEEPROM' to use the allowedSize of the EEPROM structure when checking flash memory.

- added example usage to Example2_AllFunctions

oclyke commented:
In response to #126

rustycoat commented on libraries/EEPROM/src/EEPROM.h:
```diff
@@ -69,11 +71,11 @@ Error : EEPROM start address must be divisble by 8192
         //to 2048 if needed
```
These particular comments still make it sounds like you increase the size via changing this constant and that it is limited to 2048. This line should be updated to reflect the existence of the setLength() member function, i.e. "// up to 8192 (AP3_EEPROM_MAX_LENGTH)  if needed, via the EEPROM.setLength()". (You could just cut and paste or refer to the very clear comments from Example2_AllFunctions.ino)

rustycoat commented on libraries/EEPROM/src/EEPROM.h:
```diff
@@ -174,7 +179,14 @@ struct EEPROMClass
+  }
+  void setLength(uint16_t length)
+  {
+    allowedSize = length;
```
Shouldn't this be something along the lines of:
`allowedSize = (length <= AP3_EEPROM_MAX_LENGTH) ? length : AP3_EEPROM_MAX_LENGTH;`

Otherwise I am guessing an over-range input will cause undefined behavior.

oclyke commented:
@rustycoat thanks for the comments. I've updated the notes in the header file and implemented the ternary consition in ```setLength()```. Since you were asking (?) here's how you could have made a pull request:

1. Use GitHub to "fork" our repo
2. Clone your own repo to your workspace
3. Create a new branch for your suggestions
4. Make the changes you'd like to see
5. Commit those changes to your branch on your fork
6. Open a pull request (asks to merge your branch into our repo)
7. Once the PR is accepted you can destroy your fork of the project (or keep it around for future PRs)

rustycoat commented:
Thanks @oclyke. I'm a homesteader living off grid without much technology or time for things like contributing to open source, but I greatly value the open source way of doing things. 20 years ago I was a software developer, but back then it was CVS not git and, even so, I barely remember that process :). Thanks for the instructions, hopefully I'll have some time to do my part the next time I spot a potential bug.

nseidle commented:
Looks good and tests good. Thanks for expanding this!
